velox/exec/AssignUniqueId.cpp

```diff
@@ -57,38 +60,43 @@ RowVectorPtr AssignUniqueId::getOutput() {
   if (input_ == nullptr) {
     return nullptr;
   }
-  generateIdColumn();
+  generateIdColumn(input_->size());
   auto output = fillOutput(input_->size(), nullptr);
   input_ = nullptr;
   return output;
 }
 
-void AssignUniqueId::generateIdColumn() {
-  auto numInput = input_->size();
+void AssignUniqueId::generateIdColumn(vector_size_t size) {
   auto result = results_[0];
   if (!result || !BaseVector::isReusableFlatVector(result)) {
-    result = BaseVector::create(BIGINT(), numInput, operatorCtx_->pool());
+    result = BaseVector::create(BIGINT(), size, operatorCtx_->pool());
     results_[0] = result;
   } else {
-    result->resize(numInput);
+    result->resize(size);
   }
   auto rawResults =
       result->asUnchecked<FlatVector<int64_t>>()->mutableRawValues();
 
   vector_size_t start = 0;
-  while (start < numInput) {
+  while (start < size) {
     if (rowIdCounter_ >= maxRowIdCounterValue_) {
-      rowIdCounter_ = rowIdPool_->fetch_add(kRowIdsPerRequest);
-      maxRowIdCounterValue_ =
-          std::min({rowIdCounter_ + kRowIdsPerRequest, kMaxRowId});
+      requestRowId();
     }
 
-    auto end = std::min(numInput, (int32_t)(start + kRowIdsPerRequest));
+    auto end = std::min(size, (int32_t)(start + kRowIdsPerRequest));
 
     std::iota(
         rawResults + start, rawResults + end, uniqueValueMask_ | rowIdCounter_);
     rowIdCounter_ += end;
     start = end;
   }
 }
+
+void AssignUniqueId::requestRowId() {
+  rowIdCounter_ = rowIdPool_->fetch_add(kRowIdsPerRequest);
+  maxRowIdCounterValue_ =
+      std::min({rowIdCounter_ + kRowIdsPerRequest, kMaxRowId});
+  VELOX_CHECK_LT(
```

Comment:
This is a fix for another issue, right? E.g. the loop above could got on forever without that fix. Is there a way to add test coverage for this case?

Reply:
@mbasmanova This is added to prevent rowIdCounter_ goes above 40 bit which will take 1 << 20L batches to reach. I can add a unit test in a separate PR. Not related with the current bug